orchestrator/modules/tools/execution/unified_executor.py

```
@@ -1306,10 +1313,15 @@ async def _execute_planning_tool(
                 "tool": tool_name
             }
         
-        # Ensure absolute path
+        # Ensure absolute path within workspace
         if not os.path.isabs(output_file):
             output_file = os.path.join(self.workspace_dir, output_file)
-        
+        # Validate path stays within workspace
+        workspace = _Path(self.workspace_dir).resolve()
```

Ruff (0.15.0)

[error] Undefined name `_Path`

(F821)